src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncDispatcher.cs

```diff
@@ -109,7 +114,19 @@ public async Task Start(CancellationToken cancellationToken)
                             if (Logger.IsTrace) Logger.Trace($"SyncDispatcher request: {request}, AllocatedPeer {allocation.Current}");
 
                             // Use Task.Run to make sure it queues it instead of running part of it synchronously.
-                            Task task = Task.Run(() => DoDispatch(cancellationToken, allocatedPeer, request, allocation), cancellationToken);
+                            _activeTasks.AddCount();
+                            Task task = Task.Run(
+                                () =>
+                                {
+                                    try
+                                    {
+                                        return DoDispatch(cancellationToken, allocatedPeer, request, allocation);
+                                    }
+                                    finally
+                                    {
+                                        _activeTasks.Signal();
+                                    }
+                                });
```

Comment:
Unexpected scenario, but for completion:
```suggestion
                        try
                        {
                            Task task = Task.Run(
                                () =>
                                {
                                    try
                                    {
                                        return DoDispatch(cancellationToken, allocatedPeer, request, allocation);
                                    }
                                    finally
                                    {
                                        _activeTasks.Signal();
                                    }
                                });
                            }
                            catch
                            {
                                _activeTasks.Signal();                            
                            }
```